include/semaphore.h

```diff
@@ -42,6 +42,7 @@
 #define SEM_PRIO_INHERIT          1
 #define SEM_PRIO_PROTECT          2
 #define SEM_PRIO_MASK             3
+#define SEM_PRIO_MUTEX            4
```

Comment:
```suggestion
#define SEM_TYPE_MUTEX            4
```

Reply:
@pkarashchenko here  is the patch: https://github.com/apache/nuttx/pull/8182/commits/d45ac2b157e7b9e486708ac375a44b21c4d90303